linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -623,6 +629,27 @@ void MarkXmlSchemaProvider (TypeDefinition type, CustomAttribute attribute)
 			MarkNamedMethod (type, method_name);
 		}
 
+		void MarkTypeWithDebuggerDisplayAttribute (TypeDefinition type, CustomAttribute attribute)
+		{
+			if (_context.KeepMembersForDebuggerAttributes) {
+				MarkMethods (type);
```

Comment:
I think this is just too aggressive it'll persist all methods/fields even if they are unused. You could try to evaluate content of {  } and only fallback to all if it's not simple property/field/method call